aion/web/elm/src/Ranking/View.elm

```diff
@@ -2,56 +2,116 @@ module Ranking.View exposing (..)
 import RemoteData exposing (WebData)
+import Urls exposing (host)
+import Navigation exposing (Location)
 
 
 rankingView : Model -> Html Msg
 rankingView model =
     Grid.container []
         [ Grid.row []
-            [ Grid.col [ Col.xs4 ]
-                [ h4 [] [ text "General" ] ]
+            [ Grid.col [ Col.xs4, Col.offsetXs1 ]
+                [ Form.group []
+                    [ Form.label [] [ text "Select Category" ]
+                    , Select.select [ Select.onChange OnRankingCategoryChange ] (selectCategoriesAttributes model)
+                    ]
+                ]
             ]
         , Grid.row []
-            [ Grid.col [ Col.xs4 ]
+            [ Grid.col [ Col.xs10, Col.offsetXs1 ]
                 [ rankingTable model ]
             ]
         ]
 
 
+selectCategoriesAttributes : Model -> List (Select.Item msg)
+selectCategoriesAttributes model =
+    case model.rankingData.data of
+        RemoteData.Success rankingData ->
+            List.map (\category -> Select.item [ value (toString category.categoryId) ] [ text category.categoryName] ) rankingData.rankingList
+        _ -> []
+
+
 rankingTable : Model -> Html Msg
 rankingTable model =
-    Table.simpleTable
-        ( Table.simpleThead
-            [ Table.th [] [ text "User" ]
+    Table.table
+        { options = [ Table.striped, Table.hover ]
+        , thead = Table.simpleThead
+            [ Table.th [] [ text "#" ]
+            , Table.th [ Table.cellAttr(style [ ("width", "5%") ]) ] []
+            , Table.th [] [ text "User" ]
             , Table.th [] [ text "Score" ]
             ]
-        , displayScores model.rankingData
-        )
+        , tbody = displayScores model
+        }
 
 
-displayScores : WebData RankingData -> Table.TBody Msg
-displayScores rankingData =
-    case rankingData of
+displayScores : Model -> Table.TBody Msg
+displayScores model =
+    case model.rankingData.data of
         RemoteData.Success rankingData ->
             let
-                sortedScores =
-                    List.reverse (List.sortBy .score rankingData.data)
+                categoryScores = selectedCategoryScores rankingData.rankingList model.rankingData.selectedCategoryId
             in
-                Table.tbody [] (List.map displaySingleScore sortedScores)
+                Table.tbody [] (List.map (displaySingleScore model) (sortedScoresWithIndices categoryScores))
 
         _ ->
             Table.tbody [] []
 
 
-displaySingleScore : PlayerScore -> Table.Row msg
-displaySingleScore score =
-    Table.tr []
-        [ Table.td [] [ text score.userName ]
-        , Table.td [] [ text (toString score.score) ]
-        ]
+displaySingleScore : Model -> (Int, PlayerScore) -> Table.Row msg
+displaySingleScore model indexAndScore =
+    case indexAndScore of
+        (index, playerScore) ->
+            Table.tr []
+                [ Table.td [] [ text (toString (index + 1)) ]
+                , Table.td [] (scoreImage model.location index)
```

Comment:
Maybe make `scoreImage` return `Html msg` instead of list and wrap the function call in square brackets for consistent formatting (minor insight).

Reply:
The thing is that I want to either return an empty list here or a list with one option containing an image. I am not sure how to achieve it. Let me know if you have solution for that :P

Reply:
You are right. You could try to do as in this issue: https://github.com/elm-lang/html/issues/72 or just leave as it is.

Reply:
I think `Html.none` is the way to go, but it's a subjective opinion, you can go however you want here.

Reply:
Hmm. I guess I will leave it in this shape ;)